pymc/distributions/distribution.py

```diff
@@ -861,3 +865,77 @@ def default_moment(rv, size, *rv_inputs, rv_name=None, has_fallback=False, ndim_
             f"Please provide a moment function when instantiating the {rv_name} "
             "random variable."
         )
+
+
+def change_dist_size(
+    dist: TensorVariable,
+    new_size: PotentialShapeType,
+    expand: Optional[bool] = False,
+) -> TensorVariable:
+    """Change or expand the size of a Distribution.
+
+    Parameters
+    ==========
+    dist
+        The old Distribution output.
+    new_size
+        The new size.
+    expand:
+        Expand the existing size by `new_size`.
```

Comment:
I can’t really understand the docstring for this. Could you add an example that shows what expands will do? I had to read the source code to understand it

Reply:
Sure

Reply:
I should also add a note that the original RV should not be used anymore, as the returned one will use the same RandomState/Generator variable of the original one.

Reply:
Updated the docstrings, what do you think?